src/main/java/reposense/model/RepoConfiguration.java

```diff
@@ -76,28 +76,38 @@ public static void setDatesToRepoConfigs(
      * Merges a {@code RepoConfiguration} from {@code repoConfigs} with another from {@code authorConfigs}
      * if {@code location} and {@code branch} matches.
      */
-    public static void merge(List<RepoConfiguration> repoConfigs, List<RepoConfiguration> authorConfigs) {
-        for (RepoConfiguration authorConfig : authorConfigs) {
-            if (authorConfig.location.isEmpty()) {
+    public static void merge(List<RepoConfiguration> repoConfigs, List<AuthorConfiguration> authorConfigs) {
+        for (AuthorConfiguration authorConfig : authorConfigs) {
+            if (authorConfig.getLocation().isEmpty()) {
                 for (RepoConfiguration repoConfig : repoConfigs) {
                     repoConfig.addAuthors(authorConfig.getAuthorList());
                 }
                 continue;
             }
 
-            int index = repoConfigs.indexOf(authorConfig);
+            RepoConfiguration matchingRepoConfig = getMatchingRepoConfig(repoConfigs, authorConfig);
 
-            if (index == -1) {
+            if (matchingRepoConfig == null) {
                 logger.warning(String.format(
                         "Repository %s is not found in repo-config.csv.", authorConfig.getLocation()));
                 continue;
             }
 
-            RepoConfiguration repoConfig = repoConfigs.get(index);
-            repoConfig.addAuthors(authorConfig.getAuthorList());
+            matchingRepoConfig.addAuthors(authorConfig.getAuthorList());
```

Comment:
Can change the `addAuthors` method to `setAuthorConfiguration`? 
Else we would be doing double work of setting all the values inside the `RepoConfiguration`'s `AuthorConfiguration` copy, when it was already done in `AuthorConfigCsvParser`.

Reply:
I think it is possible. One addition thing `addAuthors` does is that it propagates `IgnoreGlobList` to the authors so as long as `setAuthorConfiguration` does that it should be fine.

Reply:
Another thing is that if `RepoConfiguration` already has existing authors then they will be lost in `setAuthorConfiguration` whereas `addAuthors` just adds on to the existing authors.

Reply:
> One addition thing addAuthors does is that it propagates IgnoreGlobList to the authors so as long as setAuthorConfiguration does that it should be fine.

We can have an additional call to the propogate `ignoreGlobList` method after setting the `authorConfig`.

>Another thing is that if RepoConfiguration already has existing authors then they will be lost in setAuthorConfiguration whereas addAuthors just adds on to the existing authors.

Thing is at this stage, there should not be any authors inside `RepoConfiguration` (just parsed from `repo-config.csv`) at all, which is why we should just be setting the `AuthorConfiguration` into it. This also ensures that each `RepoConfiguration` should only have a single `AuthorConfiguration` (something is wrong if `AuthorConfigurationCsvParser` returns 2 different`AuthorConfiguration` that matches with 1 `RepoConfiguration`).

Reply:
I tried implementing this but ran into the issue below.

In `RepoConfiguration#merge`:
```
if (authorConfig.getLocation().isEmpty()) {
    for (RepoConfiguration repoConfig : repoConfigs) {
        repoConfig.addAuthors(authorConfig.getAuthorList());
    }
    continue;
}
```
If `AuthorConfiguration` has no location, the current logic will add the authors to all `repoConfigs`. I don't think we will be able to do this with `setAuthorConfiguration` because it will just override the authors.

`CsvParserTest#merge_emptyLocation_success` tests this behaviour as well so I think it is the intended behaviour.

Reply:
We can work around by removing this portion of code and putting it in another for-loop after the current for-loop, so that `setAuthorConfiguration` will not override.